test/services/SalesAtVatRateServiceSpec.scala

```diff
@@ -453,5 +428,97 @@ class SalesAtVatRateServiceSpec extends SpecBase {
         )
       }
     }
+
+    "getTotalVatOwedAfterCorrections" - {
+
+      "must return correct total when NI and EU sales exist" in {
+
+        service.getTotalVatOwedAfterCorrections(completeUserAnswers) mustBe BigDecimal(1020)
+      }
+
+      "must return zero when total NI and EU sales don't exist" in {
+
+        service.getTotalVatOwedAfterCorrections(emptyUserAnswers) mustBe BigDecimal(0)
+      }
+
+      "must return total when NI exists and EU sales don't exist" in {
+
+        service.getTotalVatOwedAfterCorrections(completeSalesFromNIUserAnswers) mustBe BigDecimal(1000)
+      }
+
+      "must return total when NI doesn't exist and EU does exist" in {
+        val answers = emptyUserAnswers
+          .set(SoldGoodsFromEuPage,true).success.value
+          .set(CountryOfSaleFromEuPage(index), Country("HR", "Croatia")).success.value
+          .set(CountryOfConsumptionFromEuPage(index, index), Country("BE", "Belgium")).success.value
+          .set(VatRatesFromEuPage(index, index), List(twentyPercentVatRate)).success.value
+          .set(NetValueOfSalesFromEuPage(index0, index0, index0), BigDecimal(100)).success.value
```

Comment:
In these tests can we use either all index or all index0. I know theyre the same, but just to make it clear theres no different indexing happening here

Reply:
done now